lib/graphql/metrics/trace.rb

```diff
@@ -0,0 +1,190 @@
+# frozen_string_literal: true
+
+module GraphQL
+  module Metrics
+    module Trace
+      def initialize(**_rest)
+        super
+        context = (@multiplex || @query).context
+        @skip_tracing = context&.fetch(GraphQL::Metrics::SKIP_GRAPHQL_METRICS_ANALYSIS, false)
```

Comment:
I'm not sure how to implement it... 

For performance's sake, each tracer is a module that's mixed into a class. Flow passes from tracer to tracer via `super` calls. (This is like `Language::Visitor`.) But it means there's no easy way to exclude one module's code from the resulting code flow 🤔

Reply:
Ah yeah I can't see a way around that. Not a big deal anyway I assume this use case is pretty specific to a library

Reply:
Oh -- there _is_ a way to do it. The application could pass in `context[:trace]` to use instead of `schema.trace_class.new`, allowing the application to customize the trace per-request. Maybe you could dress that up a bit. 🤔